scripts/Modules/mkw_core.py

```diff
@@ -88,3 +88,31 @@ def get_speed(playerIdx=0):
     xyz = math.sqrt(x ** 2 + y ** 2 + z ** 2)
 
     return speed(x, y, z, xz, xyz)
+
+# Next 3 functions are used for exact finish display
+def getIGT(lap, player): # lap is 1-indexed
+    offsets = [0xC, (player)*0x4, 0x3C, (lap-1)*0xC + 0x5]
+    value = chase_pointer(classes.getRaceInfoHolder(), offsets, 'u16')
+    return divmod(value, 0x100)
+
+def updateExactFinish(lap, player): # lap is 1-indexed
+    if player == 0:
+        address = 0x800001B0
+    elif player == 1:
+        address = 0x800001F8
+        
+    currMin, currSec = getIGT(lap, player)
+    currMil = memory.read_f32(address + (lap-1)*0x4) / 1000 % 1
+    currentLapTime = classes.timer(currMin, currSec, currMil)
+
+    pastMin, pastSec = getIGT(lap-1, player)
+    pastMil = memory.read_f32(address + (lap-2)*0x4) / 1000 % 1
+    pastLapTime = classes.timer(pastMin, pastSec, pastMil)
+
+    return currentLapTime - pastLapTime
+
+def getUnroundedTime(lap, player): # lap is 0-indexed
```

Comment:
If the lap is going to be 0-indexed in the function other people will be calling, what's the point of making the other functions 1-indexed?

Reply:
My mistake, the user function is 1-indexed (so that to find lap 1 time you pass in 1)